.snapcraft/resources/restoredb

```diff
@@ -1,72 +1,88 @@
-    echo -n "\
-*** ATTENTION ***
-* Your current database WILL BE DROPPED prior to the restore!
-* Would you like to make a backup of the current database before proceeding?
-* (y/n/Q)> "
+if [[ -d ${RESTORE_DIR}/dump ]]; then
+    rm -rf ${RESTORE_DIR}/dump
+fi
 
-    read choice
-    [[ "${choice,,}" = n* ]] && return
-    [[ "${choice,,}" = y* ]] && backupdb && return
-    exit
-}
+if [[ -f ${RESTORE_DIR}/${LOG_NAME} ]]; then
+    rm -f ${RESTORE_DIR}/${LOG_NAME}
+fi
 
-function warn {
-        echo "[!] ${1}"
-        echo "[*] Check ${restore_dir}/${log_name} for details."
-}
+BACKUP_FILE=${1}
+if [[ ! -f ${BACKUP_FILE} ]]; then
+    echo "[-] Usage: sudo snap run rocketchat-server.restoredb ${SNAP_COMMON}/rocketchat_backup_{TIMESTAMP}.tar.gz"
+    exit
+fi
 
-function abort {
-        echo "[!] ${1}"
-        echo "[*] Check ${restore_dir}/${log_name} for details."
-        echo "[-] Restore aborted!"
-        exit
-}
+if ! $(echo "${BACKUP_FILE}" | grep -q "${SNAP_COMMON}"); then
+    echo "[-] Backup file must be within ${SNAP_COMMON}."
+    exit
+fi
 
-mongo parties --eval "db.getCollectionNames()" | grep "\[ \]" >> /dev/null || ask_backup
 echo "[*] Extracting backup file..."
-restore_dir="${SNAP_COMMON}/restore"
-log_name="extraction.log"
-mkdir -p ${restore_dir}
-cd ${restore_dir}
-tar --no-same-owner --overwrite -zxvf ${backup_file} &> "${restore_dir}/${log_name}"
-[[ $? != 0 ]] && abort "Failed to extract backup files to ${restore_dir}!"
+echo "[*] Extracting backup file with \"tar --no-same-owner --overwrite -xzvf ${BACKUP_FILE}\"" &> "${RESTORE_DIR}/${LOG_NAME}"
+cd ${RESTORE_DIR}
+tar --no-same-owner --overwrite -xzvf ${BACKUP_FILE} &>> "${RESTORE_DIR}/${LOG_NAME}" \
+    || abort "Failed to extract backup files to ${RESTORE_DIR}!"
+
+if [ $(ls -l ${DATA_DIR} | wc -l) -le 1 ]; then
+    abort "No restore data found within ${DATA_DIR}!"
+fi
 echo "[*] Restoring data..."
-data_dir=$(tail "${restore_dir}/${log_name}" | grep parties/. | head -n 1)
-[[ -z ${data_dir} ]] && abort "Restore data not found within ${backup_file}!
-    Please check that your backup file contains the backup data within the \"parties\" directory."
-data_dir=$(dirname ${data_dir})
-log_name="mongorestore.log"
-mongorestore --db parties --noIndexRestore --drop ${data_dir} &> "${restore_dir}/${log_name}"
-[[ $? != 0 ]] && abort "Failed to execute mongorestore from ${data_dir}!"
-# If mongorestore.log only has a few lines, it likely didn't find the dump files
-log_lines=$(wc -l < "${restore_dir}/${log_name}")
-[[ ${log_lines} -lt 24 ]] && warn "Little or no restore data found within ${backup_file}!
-    Please check that your backup file contains all the backup data within the \"parties\" directory."
+echo "[*] Restoring data with \"${SNAP}/bin/mongorestore --db parties --noIndexRestore --drop ${DATA_DIR}\"" &>> "${RESTORE_DIR}/${LOG_NAME}"
+${SNAP}/bin/mongorestore --db parties --noIndexRestore --drop ${DATA_DIR} &>> "${RESTORE_DIR}/${LOG_NAME}" \
```

Comment:
Same here.  Lets at the very least remove the full path of the command in the echo.